nuxeo-services/nuxeo-platform-dublincore/src/main/java/org/nuxeo/ecm/platform/dublincore/service/DublinCoreStorageService.java

```diff
@@ -76,35 +78,44 @@ public void addContributor(DocumentModel doc, Event event) {
             }
         }
 
-        String[] contributorsArray = (String[]) doc.getProperty("dublincore", "contributors");
-
-        List<String> contributorsList = new ArrayList<String>();
-
-        if (contributorsArray != null && contributorsArray.length > 0) {
-            contributorsList = Arrays.asList(contributorsArray);
-            // make it resizable
-            contributorsList = new ArrayList<String>(contributorsList);
-        } else {
-            // initialize creator too
-            SchemaManager schemaMgr = Framework.getService(SchemaManager.class);
-            if (schemaMgr.getSchema("dublincore").getField("creator") != null) {
-                // First time only => creator
-                doc.setProperty("dublincore", "creator", principalName);
-            }
+        if (doc.getProperty("dublincore", "creator") == null) {
+            // First time only => creator
+            doc.setProperty("dublincore", "creator", principalName);
         }
 
+        List<String> contributorsList = getExistingContributors(doc);
         if (!contributorsList.contains(principalName)) {
             contributorsList.add(principalName);
-            String[] contributorListIn = new String[contributorsList.size()];
-            contributorsList.toArray(contributorListIn);
-            doc.setProperty("dublincore", "contributors", contributorListIn);
         }
-
+        doc.setProperty("dublincore", "contributors", contributorsList.toArray());
         doc.setProperty("dublincore", "lastContributor", principalName);
     }
 
     public void setIssuedDate(DocumentModel doc, Calendar issuedDate) {
         doc.setPropertyValue("dc:issued", issuedDate);
     }
 
+    protected List<String> getExistingContributors(DocumentModel doc) {
+        List<String> contributors = new ArrayList<>();
+        String[] contributorsArray = (String[]) doc.getProperty("dublincore", "contributors");
+        if (contributorsArray != null && contributorsArray.length > 0) {
+            List<String> sanitizedContributors = sanitizeContributors(contributorsArray);
```

Comment:
You can just `return sanitizeContributors(contributorsArray)` here and actualy inline it, and simplify:

    import org.apache.commons.lang3.ArrayUtils;

    String[] contributorsArray = (String[]) doc.getProperty("dublincore", "contributors");
    if (ArrayUtils.isEmpty(contributorsArray)) {
        return new ArrayList<>();
    }
    return Arrays.stream(contributorsArray)
                 .map(contributor -> contributor.replace(NuxeoPrincipal.PREFIX, ""))
                 .distinct()
                 .collect(Collectors.toList());

Reply:
Although I would prefer be strict for the prefix check. Write a function doing

    protected static String stripPrincipalPrefix(String principal) {
        if (principal.startswith(NuxeoPrincipal.PREFIX)) {
            return principal.substring(NuxeoPrincipal.PREFIX.length());
        } else {
            return principal;
        }
    }

Then call it with `map(DublinCoreStorageService::stripPrincipalPrefix)`